cli/internal/update/update.go

```diff
@@ -0,0 +1,138 @@
+package update
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/go-version"
+	"gopkg.in/yaml.v2"
+)
+
+var gitDescribeSuffixRE = regexp.MustCompile(`\d+-\d+-g[a-f0-9]{8}$`)
+
+// ReleaseInfo stores information about a release
+type ReleaseInfo struct {
+	VERSION     string    `json:"tag_name"`
+	URL         string    `json:"html_url"`
+	PublishedAt time.Time `json:"published_at"`
+}
+
+// StateEntry stores the information we have checked for a new version. It's
+// used to decide whether to check for a new version or not.
+type StateEntry struct {
+	CheckedForUpdateAt time.Time   `yaml:"checked_for_update_at"`
+	LatestRelease      ReleaseInfo `yaml:"latest_release"`
+}
+
+func CheckforUpdate(currentVersion, stateFilePath, repo string) (*ReleaseInfo, error) {
+
+	// stops looking for an update if it was done in the last 24 hours
+	stateEntry, _ := getStateEntry(stateFilePath)
+	if stateEntry != nil && time.Since(stateEntry.CheckedForUpdateAt).Hours() < 24 {
+		return nil, nil
+	}
+
+	releaseInfo, err := getLatestVersion(repo)
+	if err != nil {
+		return nil, err
+	}
+
+	// store the time when the check was done plus release info into a file
+	err = setStateEntry(stateFilePath, time.Now(), *releaseInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	if versionGreaterThan(releaseInfo.VERSION, currentVersion) {
+		return releaseInfo, nil
+	}
+
+	return nil, nil
+
+}
+
+func getLatestVersion(repo string) (*ReleaseInfo, error) {
+
+	req, err := http.NewRequest("GET", repo, nil)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "application/json; charset=utf-8")
+
+	client := &http.Client{Timeout: time.Second * 10}
```

Comment:
Use github token to auth the request if possible https://github.com/planetscale/cli/blob/main/internal/update/update.go#L159-L170

Reply:
it is possible. But should we store it in a .env file or is there any other approach you have in mind?

Reply:
use os.GetEnv like planet scale does, if it's not there that's fine. don't worry about .env